lc2fen/infer_pieces.py

```diff
@@ -332,56 +334,248 @@ def inferred_move(previous_fen, current_probs, changed_squares_idx):
+            if previous_list[changed_squares_idx[1]] == "P" and is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[1]
+            elif previous_list[changed_squares_idx[1]] == "p" and not is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[1]
+            elif previous_list[changed_squares_idx[2]] == "P" and is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[2]
+            elif previous_list[changed_squares_idx[2]] == "p" and not is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[2]
+            else:
+                return None
+        elif not is_empty_square(current_probs[changed_squares_idx[1]]):
+            final_sq = changed_squares_idx[1]
+            if previous_list[changed_squares_idx[0]] == "P" and is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[0]
+            elif previous_list[changed_squares_idx[0]] == "p" and not is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[0]
+            elif previous_list[changed_squares_idx[2]] == "P" and is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[2]
+            elif previous_list[changed_squares_idx[2]] == "p" and not is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[2]
+            else:
+                return None
+        elif not is_empty_square(current_probs[changed_squares_idx[2]]):
+            final_sq = changed_squares_idx[2]
+            if previous_list[changed_squares_idx[0]] == "P" and is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[0]
+            elif previous_list[changed_squares_idx[0]] == "p" and not is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[0]
+            elif previous_list[changed_squares_idx[1]] == "P" and is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[1]
+            elif previous_list[changed_squares_idx[1]] == "p" and not is_white_piece(
+                current_probs[final_sq]
+            ):
+                initial_sq = changed_squares_idx[1]
+            else:
+                return None
+        else:
+            return None
+
+        # Determine the third square
+        if sorted([initial_sq, final_sq]) == sorted(
+            [changed_squares_idx[0], changed_squares_idx[1]]
+        ):
+            third_sq = changed_squares_idx[2]
```

Comment:
Could you move these calculations of the `third_sq` into the previous if/elif chain when you set the `initial_sq`? I think the behavior should be the same and you avoid the extra comparisons (which are already done with the if/elif chain).

Reply:
Good idea and done!